src/chromap.cc

```diff
@@ -2968,12 +2987,19 @@ void Chromap<MappingRecord>::MapSingleEndReads() {
     }
   }
 
-  delete[] mm_history;
-  OutputMappingStatistics();
   std::cerr << "Mapped all reads in "
             << Chromap<>::GetRealTime() - real_start_mapping_time << "s.\n";
-  OutputMappingStatistics(num_reference_sequences, mappings_on_diff_ref_seqs_,
-                          mappings_on_diff_ref_seqs_);
+  
+  delete[] mm_history;
+  
+  OutputMappingStatistics();
+  if (!is_bulk_data_) {
+    OutputBarcodeStatistics();
+  }
+
+  index.Destroy();
+  //OutputMappingStatistics(num_reference_sequences, mappings_on_diff_ref_seqs_,
```

Comment:
Remove this.

Reply:
It's in line 2995 to be consistent with the paired-end case.

Reply:
Didn't make myself clear. Just remove the lines that have the comment.

Reply:
Done.